### ISSUE TITLE
No output from Generative Art node

### ISSUE THREAD

**Issue by GrindingAgony** [nortikin/sverchok#2275]
## No output from Generative Art node

Using the simple example file, have played with it for days, 
1. Use box in, generative art providing matrix, viewer out, load the text (which is saved as xml) - 
<rules max_depth="150">
        <rule name="entry">
            <call count="3" transforms="rz 120" rule="R1"/>
            <call count="3" transforms="rz 120" rule="R2"/>
        </rule>
        <rule name="R1">
            <call transforms="tx 2.6 rx 3.14 rz 12 ry 6 sa 0.97" rule="R1"/>
            <instance  transforms="sa 2.6" shape="box"/>
        </rule>
        <rule name="R2">
            <call transforms="tx -2.6 rz 12 ry 6 sa 0.97" rule="R2"/>
            <instance transforms="sx 2.6" shape="box"/>
        </rule>
</rules>
2. nothing out

## Expected result

vertices at least, but total blank

## Actual result

not a thing.

## Sverchok version latest, Blender 2.79

I must be missing something really obvious here - I can load .py & get them to work fine. 
I saved the text, reloaded it, even restarted Blender - blank result. Is xml usually trouble?

**Comment by enzyme69** [comment]
Have you read through this article:
http://elfnor.com/generative-art-sverchok-node-update.html

Generative Art node is a powerful node, was designed by Elfnor based on Structure Synth.

<img width="1125" alt="screen shot 2018-09-11 at 10 59 46 am" src="https://user-images.githubusercontent.com/3952687/45332075-dc05b300-b5b1-11e8-90d8-07a9efd0c87a.png">
[generative_art_002.blend.zip](https://github.com/nortikin/sverchok/files/2368956/generative_art_002.blend.zip)

Quite possible that if you supply a single Vector / Vertice, it does not generate a thing, only Matrices.

You can always plug a different object.

**Comment by GrindingAgony** [comment]
I still don't get it. I've tried everything - can you look at this file please?
![screenshot - 9_13_2018 7_13_22 am](https://user-images.githubusercontent.com/42424653/45458128-c9f85180-b724-11e8-8a98-5f32d44ec876.png)
I haven't seen so much as one vertex yet.

**Comment by enzyme69** [comment]
@GrindingAgony Yes, try clicking that "Vertices" input button. So you probably need that minimum Vector data.

**Comment by enzyme69** [comment]
And if you want to attach your blend, zip it first beforehand.

**Comment by enzyme69** [comment]
<img width="574" alt="screen shot 2018-09-18 at 10 00 21 am" src="https://user-images.githubusercontent.com/3952687/45656603-b89ab500-bb29-11e8-8f5d-09def57b776b.png">

@GrindingAgony Yes that Blend works on my SV 👍 It should work.

**Comment by GrindingAgony** [comment]
No, something is wrong, I have taken myself through about a third of the nodes & they all do what the doc says, unless I screw up. This particular node must be machine dependent in that case, or there is some graphics problem or there is a bug in that node. It will not work on my rig, all the other nodes do. When I find out what it is, I'll be back - so thanks for the help here, but there is something deeper than the obvious going on. Programming is like that. There is always something else.

**Comment by Durman** [comment]
You can copy the error message from console.
![2018-09-20_08-07-29](https://user-images.githubusercontent.com/28003269/45795363-72ba2a00-bcac-11e8-8fd3-c0e6465d7e01.png)

**Comment by GrindingAgony** [comment]
I'm getting "No module named expat; use simple XML treebuilder instead."
There has also been the occasional "no vertices data in" or something similar, but I still can't get any response.

**Comment by nortikin** [comment]
@GrindingAgony you have different python version with oter libreries.
Try other PC or even platform.

**Comment by GrindingAgony** [comment]
I am not sure how these libraries even get installed - I assumed that stuff happened along with Blender 2.79 - that is I know python is a seperate language, but what version do I need?

**Comment by nortikin** [comment]
sorry for delay
python 3.5-3.6 usually. But it is inside of blender. If you have installed python by yourself or someone installed it to you, than it shorten version or with broken XML librery. try other machine

**Comment by GrindingAgony** [comment]
Thanks - I've got time to explore this now - I'm going to do a process-of-elimination between multiple installs & different machines & let you know what I come up with. Somethings a bit wrong with my Sverchok - Back soon.

**Comment by GrindingAgony** [comment]
Well, bad news. I've spent hours trying to resolve the issue. I've used a reinstall of Blender, a new set of user prefs, (ie eliminated as much of the previous config as possible), then installed Sverchok from the latest .zip off Github, & I couldn't get any nodes to work, they're all undefined. Previous files using Sverchok do not work, then I tried this on another machine with the same result.
So - I wish I could be more helpful, but I haven't the faintest idea what's wrong.
There is something big missing here & it has been overlooked.
Programming's like that, I understand.
On both occasions I am using Windows 7, I have used a 64 bit & a 32 bit. 
Win 7 has been excellent for me, their best OS.
I haven't tried this on Win 10, but Win 10 is such a mess anyway I wouldn't bother.
Thanks.
